build.rs

```diff
@@ -76,6 +76,59 @@ impl FromStr for PythonInterpreterKind {
     }
 }
 
+struct CrossPython {
+    lib_dir: String,
+    include_dir: Option<String>,
+    os: String,
+    arch: String,
+}
+
+impl CrossPython {
+    fn both() -> Result<Self> {
+        Ok(CrossPython {
+            include_dir: Some(CrossPython::validate_variable("PYO3_CROSS_INCLUDE_DIR")?),
+            ..CrossPython::lib_only()?
+        })
+    }
+
+    fn lib_only() -> Result<Self> {
+        Ok(CrossPython {
+            lib_dir: CrossPython::validate_variable("PYO3_CROSS_LIB_DIR")?,
+            include_dir: None,
+            os: env::var("CARGO_CFG_TARGET_OS").unwrap(),
+            arch: env::var("CARGO_CFG_TARGET_ARCH").unwrap(),
+        })
+    }
+
+    fn validate_variable(var: &str) -> Result<String> {
+        let path = match env::var(var) {
```

Comment:
I think this should use `var_os`, which doesn't attempt to sanitize the variable.